### ISSUE TITLE
Remove GridSearch estimator and have it as an example with a custom metric

### ISSUE THREAD

**Issue by adrinjalali** [fairlearn/fairlearn#1374]
From what I understand, our `GridSearch` estimator can be implemented with a standard scikit-learn `GridSearchCV` while passing some custom metrics and a custom callable `refit` parameter.

I've asked @TamaraAtanasoska to take a look at it, but also nice to hear back from @MiroDudik maybe on what he thinks.

**Comment by MiroDudik** [comment]
It should be doable in principle, but there are some issues to address:
* The "grid" in the GridSearch is over the vectors of Lagrange multipliers. We do not really have any estimator that would take the vector of Lagrange multipliers as a hyperparameter vector. In other words, we would need to extract a suitable class to pass on to GridSearchCV.
* The grids are of a particular form--they try to compensate for the fact that many vectors of Lagrange mutlipliers are logically equivalent--the current grid generation routine takes care of that. That would be good to keep.

**Comment by TamaraAtanasoska** [comment]
> It should be doable in principle, but there are some issues to address:
> 
> * The "grid" in the GridSearch is over the vectors of Lagrange multipliers. We do not really have any estimator that would take the vector of Lagrange multipliers as a hyperparameter vector. In other words, we would need to extract a suitable estimator class to pass on to GridSearchCV.
> * The grids are of a particular form--they try to compensate for the fact that many vectors of Lagrange mutlipliers are logically equivalent--the current grid generation routine takes care of that. That would be good to keep.

Thanks @MiroDudik, I'll keep this in mind.
